features/pagination.feature

```diff
@@ -0,0 +1,122 @@
+Feature: Dataset Pagination
+    Background: 
+        Given I have these datasets:
+            """
+            [
+                {
+                    "id": "population-estimates"
+                },
+                {
+                    "id": "income"
+                },
+                {
+                    "id": "age"
+                }
+            ]
+            """
+    
+    Scenario: Offset skips first result of datasets when set to 1
+        When I GET "/datasets?offset=1"
+        Then I should receive the following JSON response with status "200":
+            """
+            {
+                "count":2,
+                "items": [
+                    {
+                        "id": "income"
+                    },
+                    {
+                        "id": "age"
+                    }
+                ],
+                "limit":20, 
+                "offset":1, 
+                "total_count":3
+            }
+            """
+    
+    Scenario: Results limited to 1 when limit set to 1
+        When I GET "/datasets?offset=0&limit=1"
```

Comment:
outside of this task we need to decide a standard on how much code v use case language is used in this syntax

Reply:
agreed. the initial aim was to create something like a REST API testing DSL, and we are setting precedents with the way these are written.
The step can be written in a less technical way that emphasises the DSL such as "limits" and "offsets", but that's for future dev.